src/hooks/wallets/wallets.ts

```diff
@@ -67,9 +67,13 @@ export const getSupportedWallets = (chain: ChainInfo): WalletInit[] => {
   if (window.Cypress && CYPRESS_MNEMONIC) {
     return [e2eWalletModule(chain.rpcUri)]
   }
-  return Object.entries(WALLET_MODULES)
-    .filter(([key]) => isWalletSupported(chain.disabledWallets, key))
-    .map(([, module]) => module())
+  const enabledWallets = Object.entries(WALLET_MODULES).filter(([key]) => isWalletSupported(chain.disabledWallets, key))
+
+  if (enabledWallets.length === 0) {
+    return [WALLET_MODULES.INJECTED()]
```

Comment:
MetaMask can be disabled:

![image](https://user-images.githubusercontent.com/20442784/214783246-a7aa7b57-9f05-4bb7-b746-a64c74dadc77.png)

We should probably return an empty array here, as well as filter the "recommended injected wallets" in `getRecommendedInjectedWallets`. What do you think?

Reply:
I think you're thinking too much into it. In practice, no chain will have all wallets disabled, and defaulting to the injected wallet is a quick and pragmatic solution w/o adding too many conditions IMHO.

Reply:
We should remove MetaMask from the list of wallets that can be disabled when we eventually [remove the "legacy" wallet names](https://github.com/safe-global/web-core/issues/713) then.